components/ActionSheet/src/MDCActionSheetController.m

```diff
@@ -0,0 +1,331 @@
+  _tableView.tableView.estimatedRowHeight = 56.f;
+  _tableView.tableView.rowHeight = UITableViewAutomaticDimension;
+  self.backgroundColor = [UIColor whiteColor];
+}
+
+- (void)addAction:(MDCActionSheetAction *)action {
+  [_tableView addAction:action];
+  [_tableView.tableView setNeedsLayout];
+  initialLayout = false;
+  [self.view setNeedsLayout];
+}
+
+- (NSArray<MDCActionSheetAction *> *)actions {
+  return _tableView.actions;
+}
+
+- (void)viewDidLoad {
+  [super viewDidLoad];
+  
+  if (self.header == nil) {
+    self.header = [self headerView];
+  }
+
+  [self.view addSubview:_tableView.tableView];
+  [self.view addSubview:self.header];
+}
+
+- (void)viewWillLayoutSubviews {
+  [super viewWillLayoutSubviews];
+
+  if (initialLayout == false) {
+    [self firstLayout];
+  }
+}
+
+-(void)firstLayout {
+  [self.header setNeedsLayout];
+  [self.header layoutIfNeeded];
+
+  CGFloat width = CGRectGetWidth(self.view.bounds);
+  CGFloat height = CGRectGetHeight(self.header.frame) + [_tableView tableHeightForWidth:width];
+  CGRect tableFrame = _tableView.tableView.frame;
+  tableFrame.origin.y = CGRectGetHeight(self.header.frame);
+  _tableView.tableView.frame = tableFrame;
+  self.preferredContentSize = CGSizeMake(width, height);
+  initialLayout = true;
+}
+
+- (void)viewWillAppear:(BOOL)animated {
+  [super viewWillAppear:animated];
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+  self.mdc_bottomSheetPresentationController.delegate = self;
+#pragma clang diagnostic pop
+
+  self.mdc_bottomSheetPresentationController.dismissOnBackgroundTap =
+      _transitionController.dismissOnBackgroundTap;
+  [self.view layoutIfNeeded];
+}
+
+- (BOOL)accessibilityPerformEscape {
+  if (!self.dismissOnBackgroundTap) {
+    return NO;
+  }
+  [self dismissViewControllerAnimated:YES completion:nil];
+  return YES;
+}
+
+- (void)preferredContentSizeDidChangeForChildContentContainer:(id<UIContentContainer>)container {
+  [super preferredContentSizeDidChangeForChildContentContainer:container];
+  [self.presentationController
+      preferredContentSizeDidChangeForChildContentContainer:self];
+}
+
+- (UIScrollView *)trackingScrollView {
+  return _transitionController.trackingScrollView;
+}
+
+- (void)setTrackingScrollView:(UIScrollView *)trackingScrollView {
+  _transitionController.trackingScrollView = trackingScrollView;
+}
+
+- (BOOL)dismissOnBackgroundTap {
+  return _transitionController.dismissOnBackgroundTap;
+}
+
+-(void)setDismissOnBackgroundTap:(BOOL)dismissOnBackgroundTap {
+  _transitionController.dismissOnBackgroundTap = dismissOnBackgroundTap;
+  self.mdc_bottomSheetPresentationController.dismissOnBackgroundTap = dismissOnBackgroundTap;
+}
+
+- (void)viewWillTransitionToSize:(CGSize)size withTransitionCoordinator:
+    (id<UIViewControllerTransitionCoordinator>)coordinator {
+  [super viewWillTransitionToSize:size withTransitionCoordinator:coordinator];
+  CGRect frame = self.view.frame;
```

Comment:
[minornit] Newline after invocations to super.

Reply:
Done